hal/src/nRF52840/ble_hal.cpp

```diff
@@ -15,739 +15,3147 @@
+    }
+
+    bool attMtuExchanged(uint16_t connHandle) {
+        for (int i = 0; i < attMtuList_.size(); i++) {
+            LinkAttMtu& attMtu = attMtuList_[i];
+            if (attMtu.connHandle == connHandle) {
+                return attMtu.isExchanged;
+            }
+        }
+        return true;
+    }
+
+    void clearAttMtu(uint16_t connHandle) {
+        for (int i = 0; i < attMtuList_.size(); i++) {
+            LinkAttMtu& attMtu = attMtuList_[i];
+            if (attMtu.connHandle == connHandle) {
+                attMtuList_.removeAt(i);
+            }
+        }
+    }
+
+    static uint8_t toHalCharProps(ble_gatt_char_props_t properties) {
+        uint8_t halProperties = 0;
+        if (properties.broadcast) {
+            halProperties |= BLE_SIG_CHAR_PROP_BROADCAST;
+        }
+        if (properties.read) {
+            halProperties |= BLE_SIG_CHAR_PROP_READ;
+        }
+        if (properties.write_wo_resp) {
+            halProperties |= BLE_SIG_CHAR_PROP_WRITE_WO_RESP;
+        }
+        if (properties.write) {
+            halProperties |= BLE_SIG_CHAR_PROP_WRITE;
+        }
+        if (properties.notify){
+            halProperties |= BLE_SIG_CHAR_PROP_NOTIFY;
+        }
+        if (properties.indicate) {
+            halProperties |= BLE_SIG_CHAR_PROP_INDICATE;
+        }
+        if (properties.auth_signed_wr) {
+            halProperties |= BLE_SIG_CHAR_PROP_AUTH_SIGN_WRITES;
+        }
+        return halProperties;
+    }
+
+    static void toPlatformCharProps(uint8_t halProperties, ble_gatt_char_props_t* properties) {
+        memset(properties, 0x00, sizeof(ble_gatt_char_props_t));
+        properties->broadcast = (halProperties & BLE_SIG_CHAR_PROP_BROADCAST) ? 1 : 0;
+        properties->read = (halProperties & BLE_SIG_CHAR_PROP_READ) ? 1 : 0;
+        properties->write_wo_resp = (halProperties & BLE_SIG_CHAR_PROP_WRITE_WO_RESP) ? 1 : 0;
+        properties->write = (halProperties & BLE_SIG_CHAR_PROP_WRITE) ? 1 : 0;
+        properties->notify = (halProperties & BLE_SIG_CHAR_PROP_NOTIFY) ? 1 : 0;
+        properties->indicate = (halProperties & BLE_SIG_CHAR_PROP_INDICATE) ? 1 : 0;
+        properties->auth_signed_wr = (halProperties & BLE_SIG_CHAR_PROP_AUTH_SIGN_WRITES) ? 1 : 0;
+    }
+};
+
+uint16_t GattBase::desiredAttMtu_;
+Vector<GattBase::LinkAttMtu> GattBase::attMtuList_;
 
-// Advertising duration in 10 ms units (fast mode)
-const auto FAST_ADVERT_TIMEOUT = MSEC_TO_UNITS(180000, UNIT_10_MS); // Fast mode
-const auto SLOW_ADVERT_TIMEOUT = 0; // Advertise indefinitely (slow mode)
+class BleObject {
+public:
+    class BleEventDispatcher;
+    class BleGap;
+    class Broadcaster;
+    class Observer;
+    class ConnectionsManager;
+    class GattServer;
+    class GattClient;
 
-// Minimum acceptable connection interval in 1.25 ms units
-const auto MIN_CONN_INTERVAL = MSEC_TO_UNITS(30, UNIT_1_25_MS);
+    std::unique_ptr<BleEventDispatcher> dispatcher;         /**< BLE event dispatcher. */
```

Comment:
Should these be public?

Reply:
It has to be. Because other classes, e.g. `GattClient`, need to access the `ConnectionsManager` methods via the BleObject instance, or vice versa.

Reply:
Consider adding getters in that case.